src/source/PeerConnection/SessionDescription.c

```diff
@@ -316,6 +326,64 @@ PCHAR fmtpForPayloadType(UINT64 payloadType, PSessionDescription pSessionDescrip
     return NULL;
 }
 
+/*
+ * Extracts a (hex) value after the provided prefix string. Returns true if
+ * successful.
+ */
+static BOOL readHexValue(const PCHAR input, const PCHAR prefix, PINT32 value) {
+  const PCHAR substr = STRSTR(input, prefix);
+  if (substr != NULL) {
+    if (sscanf(substr + STRLEN(prefix), "%x", value) == 1) {
+      return TRUE;
+    }
+  }
+  return FALSE;
+}
+
+/*
+ * Scores the provided fmtp string based on this library's ability to
+ * process various types of H264 streams. A score of 0 indicates an
+ * incompatible fmtp line. Beyond this, a higher score indicates more
+ * compatibility with the desired characteristics, packetization-mode=1,
+ * level-asymmetry-allowed=1, and inbound match with our preferred
+ * profile-level-id.
+ */
+INT32 getH264FmtpScore(PCHAR fmtp) {
```

Comment:
UINT32

Reply:
Interesting! Will try changing this to DOUBLE, let me know what you think. Can revert if it's too much.

Reply:
I disagree. Using float will mean that you need to come up with forward compat scheme. Whereas an integral type will allow to simply add features to get higher score and be better future proof